### PR TITLE
livepatch: fix delta check for enableByDefault

### PR DESCRIPTION
**Pull request by lucasmoura**
### Why is this needed?
On the `process_contract_deltas` function, there is a check to see if the entitlement delta has an enableByDefault directive. However, there was a typo on when fetching that value from the delta dict. We are now fixing this typo and adding an unittest for it

## Test Steps
check the new unittest for this change

## Checklist

 - [x] I have updated or added any unit tests accordingly
 - [ ] I have updated or added any integration tests accordingly
 - [ ] Changes here need to be documented, and this was done in: 

## Does this PR require extra reviews?

 - [ ] Yes
 - [x] No

### CODE REVIEW

**Review by orndorffgrant — APPROVED** [review]
Let's let CI pass, but looks good to me

File: uaclient/entitlements/tests/test_livepatch.py [hunk 1]
```diff
@@ -289,6 +289,21 @@ def test_livepatch_disable_and_setup_performed_when_resource_token_changes(
...
+        m_application_status,
+        m_enable,
+        entitlement,
+    ):
+        deltas = {"entitlement": {"obligations": {"enableByDefault": True}}}
+        m_enable.return_value = (True, None)
+        application_status = ApplicationStatus.DISABLED
+        m_application_status.return_value = (application_status, "")
+        entitlement.process_contract_deltas({}, deltas, False)
+        assert [mock.call(mock.ANY)] == m_enable.call_args_list
```
Comment (a-dubs) [inline uaclient/entitlements/tests/test_livepatch.py:305]: What does this assert actually test/confirm?

Comment (lucasmoura) [reply uaclient/entitlements/tests/test_livepatch.py:305]: It just confirm that this mocked method was actually called

**Review by dheyay — APPROVED** [review]
Looks good

CI failures on integration test builds - 
`xenial` -  are due to an extra package being held back and the apt news output check failing. Seems unrelated to the PR.

**Pull request merged**